### ISSUE TITLE
Dashboard 1.6.2 crashes on Node view

### ISSUE THREAD

**Issue by discostur** [kubernetes/dashboard#2202]
##### Steps to reproduce
Deploy Dashbaord v1.6.2 and click on "Nodes"

##### Environment
```
Installation method: kubeadm
Kubernetes version: 1.7.2
Dashboard version: v1.6.2
Commit: b7669ed2081e7b444f8db000704bd7d7f61395b3
```

##### Observed result
Dashboard reported Unknown Server Error:
```
No error data available
```

```
[2017-07-28T07:27:34Z] Incoming HTTP/1.1 GET /api/v1/node?filterBy=&itemsPerPage=15&namespace=&page=1&sortBy=d,creationTimestamp request from 10.32.0.1:64136: {}
panic: runtime error: index out of range
goroutine 164 [running]:
panic(0x1c6d8e0, 0xc420014060)
	/home/travis/.gimme/versions/go1.7.1.linux.amd64/src/runtime/panic.go:500 +0x1a1
github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster.heapsterClient.ithResourceDownload.func1(0x2cc5e60, 0xc420392390, 0x1e95537, 0x4, 0x1e97a5e, 0x6, 0xc420b9a240, 0x2, 0x2, 0xc420a93e90, ...)
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster/client.go:171 +0x5a5
created by github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster.heapsterClient.ithResourceDownload
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster/client.go:177 +0x109
```

##### Comment
That only happens with the heapster fix from 
https://github.com/kubernetes/dashboard/pull/2181/files

Without the fix i can click on Node without any problem.

**Comment by idealhack** [comment]
Duplicate of #2187 and fixed by #2193

**Comment by floreks** [comment]
@discostur could you please install and check if Nodes view works on our HEAD image?

https://github.com/kubernetes/dashboard/blob/master/docs/devel/head-releases.md

I could not reproduce it so I've added the check to prevent this error but there might be something else there.

**Comment by discostur** [comment]
@floreks 
Docker Container doesn't crash with the HEAD image, but i get the following "Internal Server Error (500)":


##### Environment
```
Installation method: 
Kubernetes version:
Dashboard version: v1.6.2
Commit: 6d8abef02d7efc9c57a157135237a8c69a3da2ac
```


##### Observed result
Dashboard reported Internal Server Error (500):
```
[restful] recover from panic situation: - runtime error: index out of range
    /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/runtime/asm_amd64.s:479
    /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/runtime/panic.go:458
    /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/runtime/panic.go:27
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster/client.go:112
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster/client.go:75
    <autogenerated>:8
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/dataselect/dataselect.go:158
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/dataselect/dataselect.go:186
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/dataselect/dataselect.go:281
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/node/list.go:84
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/node/list.go:72
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/apihandler.go:799
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/apihandler.go:441
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/container.go:267
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/filter.go:21
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/filter.go:106
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/filter.go:19
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/filter.go:79
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/filter.go:19
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/filter.go:43
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/filter.go:19
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/container.go:269
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/container.go:120
    /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/http/server.go:1726
    /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/http/server.go:2022
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/container.go:287
    /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/http/server.go:2022
    /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/http/server.go:2202
    /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/net/http/server.go:1579
    /home/travis/.gimme/versions/go1.7.1.linux.amd64/src/runtime/asm_amd64.s:2086

```

**Comment by floreks** [comment]
Could you provide me file that you used to deploy heapster? Maybe it depends on the heapster version because I don't have such error and I'm also using kubeadm to provision the cluster.

**Comment by discostur** [comment]
Sure, this is my Heapster Deployment:

```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: heapster
  namespace: kube-system
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: heapster
  namespace: kube-system
spec:
  replicas: 1
  template:
    metadata:
      labels:
        task: monitoring
        k8s-app: heapster
    spec:
      serviceAccountName: heapster
      containers:
      - name: heapster
        image: gcr.io/google_containers/heapster-amd64:v1.3.0
        imagePullPolicy: Always
        command:
        - /heapster
        - --source=kubernetes:https://kubernetes.default
        - --sink=influxdb:http://monitoring-influxdb:8086
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: heapster
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: system:heapster
subjects:
- kind: ServiceAccount
  name: heapster
  namespace: kube-system
---
apiVersion: v1
kind: Service
metadata:
  labels:
    task: monitoring
    # For use as a Cluster add-on (https://github.com/kubernetes/kubernetes/tree/master/cluster/addons)
    # If you are NOT using this as an addon, you should comment out this line.
    kubernetes.io/cluster-service: 'true'
    kubernetes.io/name: Heapster
  name: heapster
  namespace: kube-system
spec:
  ports:
  - port: 80
    targetPort: 8082
  selector:
    k8s-app: heapster
```

**Comment by floreks** [comment]
I was not able to reproduce it but I think that I have found root cause. Could you test it after #2204 gets merged and our head image will be pushed?

**Comment by discostur** [comment]
shure, just give me a short ping when https://github.com/kubernetes/dashboard/pull/2204 gets merged, then i'll test it

**Comment by floreks** [comment]
@discostur new head has been pushed.

**Comment by discostur** [comment]
@floreks yes! nodes view now works correctly ;)

However, i just found another bug but i'm not shure if this is related to that topic. If i open the new shell-exec feature (pod) then a new tab opens. If i close that shell-exec-tab, then the docker container fails (on closing the tab):

```
[2017-07-28T10:56:01Z] Incoming HTTP/1.1 GET /api/v1/pod/default/nginx-dev-2864619912-4kxf8/shell/nginx request from 10.32.0.1:58599: {}
[2017-07-28T10:56:01Z] Outcoming response to 10.32.0.1:58599 with 200 status code
E0728 10:56:10.611479       1 v2.go:105] sockjs: session not in open state
E0728 10:56:10.611479       1 v2.go:105] sockjs: session not in open state
log: exiting because of error: log: cannot create log: open /tmp/dashboard.kubernetes-dashboard-head-2870202237-hs0lb.unknownuser.log.ERROR.20170728-105610.1: no such file or directory
```

**Comment by floreks** [comment]
Great. Thanks for help!

Regarding second problem there is already issue created for it. It is however more complicated because it works on k8s `1.6.X`. In `1.7` they have changed exec behavior and also our dependencies are based on k8s `1.6`.

I have tested it on both `1.6` and `1.7` and this issue only occurs on `1.7`. Probably we will update our dependencies and get in sync with newest release. @maciaszczykm is already working on upading them. We'll see if it will make it work.

**Comment by discostur** [comment]
Thanks for your update! Looking forward for the next release ;)

**Comment by floreks** [comment]
@discostur feel free to grab updated `v1.6.3` release we've just pushed.

https://github.com/kubernetes/dashboard/releases/tag/v1.6.3

**Comment by discostur** [comment]
@floreks v.1.6.3 up and running! thanks ;)

**Comment by xuejipeng** [comment]
sorry ,I want to know if this problem has been solved, it happened again in my system.
```
E0521 07:24:48.052058       1 v2.go:105] sockjs: session not in open state
E0521 07:24:48.052058       1 v2.go:105] sockjs: session not in open state
log: exiting because of error: log: cannot create log: open /tmp/dashboard.kubernetes-dashboard-7568d694bc-bnnt5.unknownuser.log.ERROR.20180521-072448.1: no such file or directory
```
my dashborad version is 1.8.3, kubernetes version is 1.8.6 and heapster version os 1.5.0

**Comment by xuejipeng** [comment]
@floreks

**Comment by xuejipeng** [comment]
sorry，I have solved it, when i mount /tmp to pod, it work for me.
hope to help more people
 ```   volumeMounts:
        - mountPath: /tmp
          name: tmp-volume
   volumes:
      - name: tmp-volume
        emptyDir: {}
```
